src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationIconValueProvider.cs

```diff
@@ -47,10 +45,23 @@ public ApplicationIconValueProvider(UnconfiguredProject unconfiguredProject, IFi
             // This isn't a limitation of ApplicationIcon; it is to simply mimic the legacy property page interaction.
             if (_unconfiguredProject.IsOutsideProjectDirectory(unevaluatedPropertyValue))
             {
-                string fileName = Path.GetFileName(unevaluatedPropertyValue);
-                string destination = Path.Combine(Path.GetDirectoryName(_unconfiguredProject.FullPath), fileName);
-                _fileSystem.CopyFile(unevaluatedPropertyValue, destination, overwrite: true);
-                return fileName;
+                try
+                {
+                    string fileName = Path.GetFileName(unevaluatedPropertyValue);
+                    var destinationInfo = new FileInfo(Path.Combine(Path.GetDirectoryName(_unconfiguredProject.FullPath), fileName));
+                    if (destinationInfo.Exists && destinationInfo.IsReadOnly)
+                    {
+                        // https://stackoverflow.com/a/8081331/294804
+                        File.SetAttributes(destinationInfo.FullName, destinationInfo.Attributes & ~FileAttributes.ReadOnly);
```

Comment:
I don't think this is safe to do. For example, if using TFS, the file may be marked as read-only because it is currently not checked out.

I'd suggest using your failure path if the target file exists and is read only. Ideally the user would see a message explaining why it failed, but we don't have a good way of surfacing such issues from interceptors.

Reply:
Done.